Source/ACE.Entity/AceObject.cs

```diff
@@ -1817,8 +1841,7 @@ public object Clone(uint guid)
             AceObject ret = (AceObject)Clone();
             ret.AceObjectId = guid;
             // We are cloning a new AceObject with a new AceObjectID - need to set this to false. Og II
-            ret.HasEverBeenSavedToDatabase = false;
-            ret.IsDirty = true;
+            ret.SetDirtyFlags();
```

Comment:
HasEverBeenSavedToDatabase possible issue in removing this..

is it possible for something to be HasEverBeenSavedToDatabase == true and IsDirty == false? I'd assume something could be dirty that already has been saved to the database, so I don't follow the change entirely.